client.go

```diff
@@ -0,0 +1,60 @@
+package main
+
+import (
+	"bufio"
+	"net"
+
+	baps3 "github.com/UniversityRadioYork/baps3-go"
+)
+
+type Client struct {
+	conn       net.Conn
+	reader     *bufio.Reader
+	writer     *bufio.Writer
+	listenerCh chan []byte
+	Outgoing   chan []byte
+}
+
+func MakeClient(conn net.Conn, clientCh chan []byte) *Client {
+	client := &Client{
+		conn:       conn,
+		reader:     bufio.NewReader(conn),
+		writer:     bufio.NewWriter(conn),
+		listenerCh: clientCh,
+		Outgoing:   make(chan []byte),
+	}
+
+	go client.Read()
```

Comment:
This currently never terminates, unless I'm missing something obvious. There should be a way to gracefully kill the goroutines for a connection